internal/resource/workflowGroups/model.go

```diff
@@ -0,0 +1,98 @@
+	sgsdkgo "github.com/StackGuardian/sg-sdk-go"
+	flatteners "github.com/StackGuardian/terraform-provider-stackguardian/internal/flattners"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type WorkflowGroupResourceModel struct {
+	Organization types.String `tfsdk:"organization"`
+	ResourceName types.String `tfsdk:"resource_name"`
+	Description  types.String `tfsdk:"description"`
+	IsActive     types.String `tfsdk:"is_active"`
+	Tags         types.List   `tfsdk:"tags"`
+}
+
+func (m *WorkflowGroupResourceModel) ToAPIModel(ctx context.Context) (*sgsdkgo.WorkflowGroup, diag.Diagnostics) {
+	diag := diag.Diagnostics{}
+	apiModel := sgsdkgo.WorkflowGroup{
+		ResourceName: m.ResourceName.ValueStringPointer(),
+		Description:  m.Description.ValueStringPointer(),
+		IsActive:     (*sgsdkgo.IsArchiveEnum)(m.IsActive.ValueStringPointer()),
+	}
+
+	// Convert Tags from types.List to []string
+	elements := make([]types.String, 0, len(m.Tags.Elements()))
+	diags := m.Tags.ElementsAs(ctx, &elements, false)
+	diag.Append(diags...)
+	if diag.HasError() {
+		return nil, diag
+	}
+	var tags []string
+	for _, tag := range elements {
+		tags = append(tags, tag.ValueString())
+	}
+
+	apiModel.Tags = tags
+
+	return &apiModel, nil
+}
+
+func (m *WorkflowGroupResourceModel) ToPatchedAPIModel(ctx context.Context) (*sgsdkgo.PatchedWorkflowGroup, diag.Diagnostics) {
+	diag := diag.Diagnostics{}
+	apiModel := sgsdkgo.PatchedWorkflowGroup{
+		ResourceName: m.ResourceName.ValueStringPointer(),
+		Description:  m.Description.ValueStringPointer(),
+		IsActive:     (*sgsdkgo.IsArchiveEnum)(m.IsActive.ValueStringPointer()),
+	}
+
+	// Convert Tags from types.List to []string
+	elements := make([]types.String, 0, len(m.Tags.Elements()))
+	diags := m.Tags.ElementsAs(ctx, &elements, false)
+	diag.Append(diags...)
+	if diag.HasError() {
+		return nil, diag
+	}
+	var tags []string
+	for _, tag := range elements {
+		tags = append(tags, tag.ValueString())
+	}
+
+	apiModel.Tags = tags
+
+	return &apiModel, nil
+}
+
+func buildAPIModelToWorkflowGroupModel(apiResponse *sgsdkgo.WorkflowGroup) (*WorkflowGroupResourceModel, diag.Diagnostics) {
+	diag := diag.Diagnostics{}
+	WorkflowGroupModel := &WorkflowGroupResourceModel{
+		ResourceName: flatteners.String(*apiResponse.ResourceName),
+		Description:  flatteners.String(*apiResponse.Description),
+		IsActive:     flatteners.String(string(*apiResponse.IsActive.Ptr())),
+	}
+
+	//The response field 'org' is in the format '/orgs/<org-name>' but the input value in just <org-name>
+	org := strings.ReplaceAll(apiResponse.OrgId, "/orgs/", "")
+	WorkflowGroupModel.Organization = flatteners.String(org)
+
+	// Convert Tags from []string to types.List
+	var tags []attr.Value
+	for _, tag := range apiResponse.Tags {
+		tags = append(tags, flatteners.String(tag))
+	}
+	tagsList, diags := types.ListValue(types.StringType, tags)
```

Comment:
types.ListFromValue is a better option than types.ListValue since it takes context.